test/services/open_banking_connector_spec.rb

```diff
@@ -21,25 +23,59 @@ class FetchMovementsTest < ActiveSupport::TestCase
 
       ::Nordigen::NordigenClient.stub(:new, @client_mock) do
         exception = assert_raises(StandardError) do |e|
-          OpenBankingConnector.new.fetch_movements(external_account_id: "1")
+          OpenBankingConnector.new.fetch_movements(account: @account_instance)
         end
 
         assert_equal exception.message, error.to_json
       end
     end
 
-    test "returns the main balance of the given existent account as a float number, the currency and the transactions" do
+    test "When no moments are present, then returns the main balance of the given existent account as a float number, the currency and the transactions for last 4 months" do
+      date_from = 4.months.ago.strftime('%F')
+      date_to = @current_date
+
       balances_response = {"balances" => [{"balanceAmount" => {"amount" => "896.13", "currency" => "EUR"}, "balanceType" => "information", "lastChangeDateTime" => "2023-10-12T00:00:00Z"}, {"balanceAmount" => {"amount" => "935.37", "currency" => "EUR"}, "balanceType" => "interimBooked", "lastChangeDateTime" => "2023-10-12T00:00:00Z"}]}
+      account_mock = Minitest::Mock.new
+      account_mock.expect(:get_balances, balances_response.deep_symbolize_keys)
+      account_mock.expect(:get_transactions, {"transactions" => {"booked" => MockClient::Stubs.transactions}}, date_from:, date_to:)
+
+      @client_mock.expect(:account, account_mock, ["1"])
+
+      ::Nordigen::NordigenClient.stub(:new, @client_mock) do
+        response = OpenBankingConnector.new.fetch_movements(account: @account_instance)
 
+        assert_equal @account_instance.movements.exists?, false
+        assert_equal response.balance, 896.13
+        assert_equal response.currency, "EUR"
+
+        assert_equal response.transactions.first[:id], 0
+        assert_equal response.transactions.first[:amount], 100.0
+        assert_equal response.transactions.first[:date], "2023-07-24"
+        assert_equal response.transactions.first[:description], "Pago en EL CORTE INGLES"
+
+        assert_equal response.transactions.second[:id], 1
+        assert_equal response.transactions.second[:amount], 200.0
+        assert_equal response.transactions.second[:date], "2023-07-24"
+        assert_equal response.transactions.second[:description], "Pago en EL CORTE INGLES"
+      end
+    end
+
+    test 'When moments are present, then returns the main balance of the given existent account as a float number, the currency and the transactions from last moment onwards' do
```

Comment:
There's a slight typo here, instead of moments it should be movement, would you mind amending it?

Reply:
Sure, on it.